07-errors.md

```diff
@@ -0,0 +1,311 @@
+# Error Conditions
+
+This document specifies the error conditions in the FVM and the associated error codes/numbers.
+
+## Exit Codes
+
+Exit codes are numeric, 32bit integer codes that signal the outcome of the execution.
+
+A 0 exit code is an _implied_ exit code: it is generated when the actor returns normally. Non-zero exit code are either:
+
+1. _explicitly_ signalled by an actor through an early `abort` operation
+2. _explicitly_ generated by the VM because it encountered an abnormality during execution
+
+In both cases, a non-zero exit code indicates that execution was interrupted and that the actors
+state was left unchanged.
+
+Exit codes appear on-chain inside the message receipt, and therefore form part of consensus.
+
+The only exit code returned on success is `0`, or `Ok`, indicating that the message was applied
+successfully.
+
+All non-zero exit codes indicate that the message _was not_ applied and that the chain state _was
+not_ updated.
+
```

Comment:
nit: In case we want to add something on this, as it's related, and I'm sure others will trip over this.

```suggestion

On a related note, non-zero exit codes and return data have a XOR relationship with one another.
- On a `0` exit code, the actor can return data.
- On a non-zero exit code, the actor cannot return data.
```

Reply:
I've removed the text (we were repeating ourselves) and have updated the text right above this to describe how exit codes relate to return values.